### PR TITLE
[Web UI] Network error banner

### PR DESCRIPTION
**Pull request by dabria**
## What is this change?
Adds a banner which can update LocalNetwork retry to `true`. 

## Why is this change necessary?
Will give a user the ability to attempt a reconnect if they lose connection

## Does your change need a Changelog entry?
No

### CODE REVIEW

File: dashboard/src/components/relocation/RetryConnectionBanner.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { withApollo } from "react-apollo";
+import Button from "@material-ui/core/Button";
+import Banner from "./Banner";
+import retryLocalNetwork from "../../mutations/retryLocalNetwork";
+
+class RetryConnectionBanner extends React.PureComponent {
+  static propTypes = { client: PropTypes.object.isRequired };
+
+  retryConnection = () => {
+    console.log("this is running");
```
Comment (10xjs) [inline dashboard/src/components/relocation/RetryConnectionBanner.js]: Remove console log

File: dashboard/src/components/relocation/RetryConnectionBanner.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    retryLocalNetwork(this.props.client);
+  };
+
+  render() {
+    return (
+      <Banner
+        message="You've lost network connection."
+        variant="warning"
+        actions={
+          <Button color="inherit" onClick={() => this.retryConnection()}>
```
Comment (10xjs) [inline dashboard/src/components/relocation/RetryConnectionBanner.js]: Set this button to disabled when `retry` is true to provide subtle feedback that clicking did something

File: dashboard/src/components/relocation/RetryConnectionBanner.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  retryConnection = () => {
+    console.log("this is running");
+    retryLocalNetwork(this.props.client);
+  };
+
+  render() {
+    return (
+      <Banner
+        message="You've lost network connection."
```
Comment (10xjs) [inline dashboard/src/components/relocation/RetryConnectionBanner.js]: We could potentially provide a better explanation here. Maybe a link to a help document if one exists (would probably have to create that).

There's even an opportunity to use the `NetworkInformation` API (https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation) which @jamesdphillips was interested in.

File: dashboard/src/components/relocation/RetryConnectionBanner.js [hunk 4]
```diff
@@ -1,25 +1,35 @@
...
+        // TODO: make message better by linking to a troubleshooting guide
+        // or use NetworkInformation API: https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation
+        message="You've lost network connection. Check your network connection or the server."
         variant="warning"
         actions={
-          <Button color="inherit" onClick={() => this.retryConnection()}>
+          <Button
+            color="inherit"
+            onClick={() => this.retryConnection()}
+            disabled={data.localNetwork.offline}
```
Comment (10xjs) [inline dashboard/src/components/relocation/RetryConnectionBanner.js]: `data.localNetwork.retry`

Comment (dabria) [reply dashboard/src/components/relocation/RetryConnectionBanner.js]: Oh frick haha. Thanks

**Pull request merged**